common/services/settings.ts

```diff
@@ -130,3 +130,21 @@ export function formatLabelValuePair(label, value){
 		? `${value} (${label})`
 		: `${value}`
 };
+
+/**
+ * Get the configuration value if the customization is enabled.
+ * @param options
+ * @returns
+ */
+export function getSettingDependOnCustomizationIsEnabled(configuration: any, settingKey: string, overwriteDefaultValue?: any) {
+  const isCustomizationEnabled = typeof configuration['customization.enabled'] === 'undefined'
+    ? getSettingDefaultValue('customization.enabled')
+    : configuration['customization.enabled'];
+  const defaultValue = typeof overwriteDefaultValue !== 'undefined'
+    ? overwriteDefaultValue
+    : getSettingDefaultValue(settingKey);
+
+	return ( isCustomizationEnabled && settingKey.startsWith('customization') && settingKey !== 'customization.enabled')
+		? (configuration[settingKey] ?? defaultValue)
+		: defaultValue;
```

Comment:
I have several suggestions for improvement for this function:

1. The name. It is too long because it gives information that doesn't matter to the user. The goal (as I understand it) is to get from the `configuration` "object" the option given by `settingKey` (as simple as a getter). The `dependOnCustomizationIsEnabled` is an implementation detail of the function. I, as a programmer, want to get that setting, how the function gets it doesn't matter to me. It breaks the abstraction a bit in my opinion and leaves a rather unattractive name.

2. The types. I understand that `configuration` should be at best a type defined by us, or at least an object. On the other hand, the `overwriteDefaultValue` sounds like a boolean to me, but... then, looking at the code, I see that it seems to be the value that I want the setting to have (?). Even looking at the code, I don't see what it does. Finally, it is not defined what the function returns (the type of the return value is missing)

3. Ternary operator. It's ok to use it in simple situations, but not when it makes the code more messy and less understandable. It is part of our [STYLEGUIDE](https://github.com/wazuh/wazuh-kibana-app/blob/master/STYLEGUIDE.md#only-use-ternary-operators-for-small-simple-code). From my point of view, less code is not always better, it is preferable to have a few more lines if we gain readability.

4. Documentation. It doesn't do much good to add a documentation header to the function if it is not properly filled in, or if it is not updated if we modify the code. In the end, we have documentation that says the code does one thing when it is really doing something else. It just adds more mess and more wasted time to understand it when we go back through this block of code.

Reply:
Solved in https://github.com/wazuh/wazuh-kibana-app/pull/4507/commits/7ad0476f4ff038fe37195536ecdb4416d29b6c60